### PR TITLE
chore: add zendesk-dev.com domain to allow for scooter

### PR DESCRIPTION
**Pull request by zFlabmonsta**
:v:

/cc @zendesk/vegemite

### Description
This PR contains whitelisting scooter domain to the allow list.

### Tasks
- [ ] Include comments/inline docs where appropriate
- [ ] Add [unit tests](/spec)
- [ ] Update changelog [here](https://github.com/zendesk/zaf_docs/blob/master/doc/v2/dev_guide/changelog.md)
- [ ] Add label from one of 'patch'/'minor'/'major' if you want to create a release when merged

### References
* JIRA: https://zendesk.atlassian.net/browse/VEG-XXX

### DevQA Steps

- Make sure [Zendesk Apps Framework regression tests](https://zendesk.atlassian.net/wiki/spaces/ENG/pages/641702848/DevQA+process+in+Vegemite#DevQAprocessinVegemite-regression-checks) are passing with this change.
- Add specific DevQA instructions here ....

NOTE: DevQA steps are to be actioned only once code has been reviewed and approved.

### Risks
* [HIGH | medium | low] Does it work across browsers (including IE!)?
* [HIGH | medium | low] Does it work in the different products (Support, Chat, Connect)?
* [HIGH | medium | low] Are there any performance implications?
* [HIGH | medium | low] Any security risks?
* [HIGH | medium | low] What features does this touch?

### Rollback Plan

1. Quickly [roll back] to the prior release so that customers can refresh to resolve their issue.
1. Revert this PR to restore the master branch to a deployable green state.
1. Notify the author.

[roll back]: https://github.com/zendesk/zendesk_app_framework_sdk/blob/master/DEPLOY.md#recovery

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -283,6 +283,7 @@ function isOriginValid (origin) {
       /^https:\/\/.+\.zd-dev\.com$/,
       /^https:\/\/.+\.zd-master\.com$/,
       /^https:\/\/.+\.zendesk-staging\.com$/,
+      /^https:\/\/.+\.zendesk-dev\.com$/,
```
Comment (zFlabmonsta) [inline lib/client.js:286]: we can further decrease the scope to specifc region. Currently there are 2 regions 

`.euc1.scooter.zendesk-dev.com`
`.usw2.scooter.zendesk-dev.com`

and a third in apac is on its way. https://zendesk.atlassian.net/browse/PLAN-7053

Comment (Shengming-Yan) [reply lib/client.js:286]: Domain level looks fine to me

**Review by Shengming-Yan — APPROVED** [review]

File: lib/client.js [hunk 2]
```diff
@@ -283,6 +283,7 @@ function isOriginValid (origin) {
       /^https:\/\/.+\.zd-dev\.com$/,
       /^https:\/\/.+\.zd-master\.com$/,
       /^https:\/\/.+\.zendesk-staging\.com$/,
+      /^https:\/\/.+[euc1|usw2]\.scooter\.zendesk-dev\.com$/,
```
Comment (Shengming-Yan) [inline lib/client.js:286]: Sorry, it seems my previous comment make you confusion. I was trying to say domain level(zendesk-dev.com) looks good enough. Not need to specify into subdomain level.

**Review by anushkafka — APPROVED** [review]

File: lib/client.js (hunk 2 above)
Comment (zFlabmonsta) [reply lib/client.js:286]: Ah ok. I did see the comment from before 😅

**Pull request merged**